src/broadcast/helpers.js

```diff
@@ -57,14 +57,94 @@ exports = module.exports = steemBroadcast => {
           break;
         }
       }
+
       // user does not exist in authorized list
       if (totalAuthorizedUser === updatedAuthority.account_auths.length) {
         return cb(null, null);
       }
 
-      const owner = role === "owner" ? updatedAuthority : undefined;
-      const active = role === "active" ? updatedAuthority : undefined;
-      const posting = role === "posting" ? updatedAuthority : undefined;
+      const owner = role === 'owner' ? updatedAuthority : undefined;
+      const active = role === 'active' ? updatedAuthority : undefined;
+      const posting = role === 'posting' ? updatedAuthority : undefined;
+
+      steemBroadcast.accountUpdate(
+        activeWif,
+        userAccount.name,
+        owner,
+        active,
+        posting,
+        userAccount.memo_key,
+        userAccount.json_metadata,
+        cb
+      );
+    });
+  };
+
+  steemBroadcast.addKeyAuth = (
+    activeWif,
+    username,
+    authorizedKey,
+    role = 'posting',
+    weight = 1,
+    cb
+  ) => {
+    api.getAccountsAsync([username]).then(([userAccount]) => {
```

Comment:
If this fails the callback won't be called, you want to use one of either promises or callbacks all the way